DS_and_Algorithm/Selenium/test.java

```diff
@@ -0,0 +1,706 @@
+    driver.get(pages.formPage);
+
+    WebElement originallySelected = enabledSelectedRadioButton();
+    assertSelected(originallySelected);
+
+    WebElement toSelect = enabledUnselectedRadioButton();
+    assertNotSelected(toSelect);
+
+    toSelect.click();
+    assertNotSelected(originallySelected);
+    assertSelected(toSelect);
+  }
+
+  @Test
+  void testShouldBeAbleToToggleAnEnabledUnselectedCheckbox() {
+    driver.get(pages.formPage);
+    assertCanToggle(enabledUnselectedCheckbox());
+  }
+
+  @Test
+  void testShouldBeAbleToToggleAnEnabledSelectedCheckbox() {
+    driver.get(pages.formPage);
+    assertCanToggle(enabledSelectedCheckbox());
+  }
+
+  @Test
+  void testClickingOnASelectedRadioButtonShouldLeaveItSelected() {
+    driver.get(pages.formPage);
+
+    WebElement button = enabledSelectedRadioButton();
+    assertThat(button.isSelected()).isTrue();
+
+    button.click();
+
+    assertThat(button.isSelected()).isTrue();
+  }
+
+  @Test
+  void testShouldBeAbleToToggleEnabledMultiSelectOption() {
+    driver.get(pages.formPage);
+    assertCanToggle(selectedMultipleSelectOption());
+  }
+
+  @Test
+  void testShouldBeAbleToToggleSelectableCheckboxByClickingOnIt() {
+    driver.get(pages.formPage);
+
+    WebElement checkbox = enabledUnselectedCheckbox();
+    assertNotSelected(checkbox);
+
+    checkbox.click();
+    assertSelected(checkbox);
+
+    checkbox.click();
+    assertNotSelected(checkbox);
+  }
+
+  @Test
+  void testShouldBeAbleToSelectSelectableRadioButtonByClickingOnIt() {
+    driver.get(pages.formPage);
+
+    WebElement radioButton = enabledUnselectedRadioButton();
+    assertNotSelected(radioButton);
+
+    radioButton.click();
+    assertSelected(radioButton);
+
+    radioButton.click();
+    assertSelected(radioButton);
+  }
+
+  @Test
+  void testClickingDisabledSelectedCheckboxShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(randomlyDisabledSelectedCheckbox());
+  }
+
+  @Test
+  void testClickingDisabledUnselectedCheckboxShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledUnselectedCheckbox());
+  }
+
+  @Test
+  void testClickingDisabledSelectedRadioButtonShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledSelectedRadioButton());
+  }
+
+  @Test
+  void testClickingDisabledUnselectedRadioButtonShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledUnselectedRadioButton());
+  }
+
+  private void assertNotSelected(WebElement element) {
+    assertSelected(element, UNSELECTED);
+  }
+
+  private void assertSelected(WebElement element) {
+    assertSelected(element, SELECTED);
+  }
+
+  private void assertSelected(WebElement element, boolean isSelected) {
+    wait.until(ExpectedConditions.elementSelectionStateToBe(element, isSelected));
+    assertThat(element.isSelected())
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/17#issuecomment-2188976663>#automa</a><span> - 06/25/2024, 01:40 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The method 'assertSelected' uses a chained assertion with 'assertThat' which is split across multiple lines, making it harder to read.
</div>
<div id="fix">
<b>Fix:</b> Refactor the assertion to improve readability by consolidating the chained calls.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -391,7 +391,7 @@
 -    assertThat(element.isSelected())
 -        .describedAs(
 -            "Expected element %s to be %s",
 -            describe(element), selectedToString(isSelected), selectedToString(!isSelected))
 -        .isEqualTo(isSelected);
 +    assertThat(element.isSelected()).describedAs(
 +        "Expected element %s to be %s",
 +        describe(element), selectedToString(isSelected), selectedToString(!isSelected)
 +    ).isEqualTo(isSelected);
```
</div>
</details>
</div>
<br>


</div>